### ISSUE TITLE
Ongoing WebGL related dev

### ISSUE THREAD

**Issue by almarklein** [bokeh/bokeh#2590]
This is a meta-issue to keep tack of the tasks related to support for WebGL in Bokeh.

Tasks:

* [x] Set up system by which Bokeh can let glyphs draw themselves in WebGL and have the result blended with the normal canvas.
* [x] Create example to better show off WebGL: @bryevdv suggested a variant of a [sklearn clustering example](http://scikit-learn.org/stable/auto_examples/cluster/plot_cluster_comparison.html) (#2604)
* [x] Make WebGL support work on Safari
* [x] See if we can make plots share a gl context to allow 16+ plots on one page (#2607)
* [x] See if we can still support selections with > 65k points by drawing in multiple passes
* [ ] Use memory buffers instead of normal arrays for data throughout BokehJS (#2204)
* [ ] Track down data processing bottlenecks in BokehJS that prevent taking full advantage of WebGL
* [ ] Maybe support some nonlinear transformations in WebGL?

Supported glyphs:

* [x] line (#2681)
* [ ] images
* [ ] maps
* [ ] ...
* markers:
  * [x] circle
  * [x] square
  * [x] cross
  * [x] x
  * [x] circle x
  * [x] square cross
  * [x] diamond
  * [x] diamond cross
  * [x] triangle
  * [x] inverted triangle

**Comment by bryevdv** [comment]
I think this can be closed. Things remaining have their own issues.
